test/e2e/framework/network_pods.go

```diff
@@ -72,17 +71,58 @@ func (f *Framework) CreateTCPCheckConnectorPod(cluster int, remoteCheckPod *v1.P
 					// resource environments from not sending at least some data before timeout.
 					Command: []string{"sh", "-c", "for in in $(seq 50); do echo connector says $SEND_STRING; done | nc -v $REMOTE_IP $REMOTE_PORT -w 5 >/dev/termination-log 2>&1"},
 					Env: []v1.EnvVar{
-						{Name: "REMOTE_PORT", Value: strconv.Itoa(TestPort)},
-						{Name: "SEND_STRING", Value: sendString},
-						{Name: "REMOTE_IP", Value: remoteIP},
+						{Name: "REMOTE_PORT", Value: strconv.Itoa(tp.Config.Port)},
+						{Name: "SEND_STRING", Value: tp.Config.Data},
+						{Name: "REMOTE_IP", Value: tp.Config.RemoteIP},
 					},
 				},
 			},
 		},
 	}
 
-	pc := f.ClusterClients[cluster].CoreV1().Pods(f.Namespace)
-	tcpCheckPod, err := pc.Create(&tcpCheckConnectorPod)
+	pc := tp.framework.ClusterClients[tp.Config.Cluster].CoreV1().Pods(tp.framework.Namespace)
+	var err error
+	tp.Pod, err = pc.Create(&tcpCheckConnectorPod)
 	Expect(err).NotTo(HaveOccurred())
-	return tcpCheckPod
+}
+
+func nodeAffinity(scheduling TestPodScheduling) *v1.Affinity {
+
+	const gatewayLabel = "submariner.io/gateway"
```

Comment:
If all the nodes in the cluster are labeled as GatewayCapableNodes (i.e., submariner.io/gateway), then this test would fail right. So, can we do something to avoid this issue?